init.go

```diff
@@ -25,193 +25,39 @@ const (
 )
 
 var (
+	services apid.Services
 	log      apid.LogService
-	data     apid.DataService
-	events   apid.EventsService
-	unsafeDB apid.DB
-	dbMux    sync.RWMutex
 )
 
-func getDB() apid.DB {
-	dbMux.RLock()
-	db := unsafeDB
-	dbMux.RUnlock()
-	return db
-}
-
-func setDB(db apid.DB) {
-	dbMux.Lock()
-	unsafeDB = db
-	dbMux.Unlock()
-}
-
 func init() {
 	apid.RegisterPlugin(initPlugin)
 }
 
-func initPlugin(services apid.Services) (apid.PluginData, error) {
+func initPlugin(s apid.Services) (apid.PluginData, error) {
+	services = s
+
 	log = services.Log().ForModule("apidVerifyAPIKey")
 	log.Debug("start init")
 
-	data = services.Data()
-	events = services.Events()
+	log = services.Log()
+	dbMan := &dbManager{
+		data:  services.Data(),
+		dbMux: sync.RWMutex{},
```

Comment:
It would be better to use `*RWMutex` instead of `RWMutex`, in case we copy (pass by value) it unintentionally.
Try this: `&sync.RWMutex{}`
According to https://golang.org/src/sync/rwmutex.go?s=894:1226#L18
"An RWMutex must not be copied after first use."